threedi_schema/migrations/versions/0300_upgrade_db_settings.py

```diff
@@ -0,0 +1,296 @@
+        ("initial_waterlevel", "initial_water_level"),
+        ("water_level_ini_type", "initial_water_level_aggregation"),
+        ("initial_waterlevel_file", "initial_water_level_file"),
+    ],
+    "physical_settings": [
+        ("advection_1d", "use_advection_1d"),
+        ("advection_2d", "use_advection_2d"),
+    ],
+    "interception": [
+        ("interception_global", "interception"),
+        ("interception_file", "interception_file"),
+    ],
+}
+
+# Columns set to True if a mapping between use_* and settings table exists
+# (boolean column, setting id, setting table)
+GLOBAL_SETTINGS_ID_TO_BOOL = [
+    ("use_groundwater_storage", "groundwater_settings_id", "groundwater"),
+    ("use_interflow", "interflow_settings_id", "interflow"),
+    ("use_structure_control", "control_group_id", "v2_control_group"),
+    ("use_simple_infiltration", "simple_infiltration_settings_id", "simple_infiltration"),
+    ("use_vegetation_drag_2d", "vegetation_drag_settings_id", "vegetation_drag_2d"),
+]
+
+
+def rename_tables(table_sets: List[Tuple[str, str]]):
+    # no checks for existence are done, this will fail if a source table doesn't exist
+    for src_name, dst_name in table_sets:
+        op.rename_table(src_name, dst_name)
+
+
+def rename_columns(table_name: str, columns: List[Tuple[str, str]]):
+    # no checks for existence are done, this will fail if table or any source column doesn't exist
+    with op.batch_alter_table(table_name) as batch_op:
+        for src_name, dst_name in columns:
+            batch_op.alter_column(src_name, new_column_name=dst_name)
+
+
+def make_all_columns_nullable(table_name, id_name: str = 'id'):
+    # no checks for existence are done, this will fail if table doesn't exist
+    connection = op.get_bind()
+    table = sa.Table(table_name, sa.MetaData(), autoload_with=connection)
+    with op.batch_alter_table(table_name) as batch_op:
+        for column in table.columns:
+            if column.name == id_name:
+                continue
+            batch_op.alter_column(column_name=column.name, nullable=True)
+
+
+def create_new_tables(table_names: List[str]):
+    # no checks for existence are done, this will fail if any table already exists
+    for table_name in table_names:
+        op.create_table(table_name, sa.Column("id", sa.Integer(), primary_key=True))
+
+
+def add_columns_to_tables(table_columns: List[Tuple[str, Column]]):
+    # no checks for existence are done, this will fail if any column already exists
+    for dst_table, col in table_columns:
+        with op.batch_alter_table(dst_table) as batch_op:
+            batch_op.add_column(col)
+
+
+def move_multiple_values_to_empty_table(src_table: str, dst_table: str, columns: List[str]):
+    # move values from one table to another
+    # no checks for existence are done, this will fail if any table or column doesn't exist
+    dst_cols = ', '.join(dst for _, dst in columns)
```

Comment:
Maybe I'm being overcautious, but it might be best to surround column names (when used directly in SQL) with double quotes, to prevent it being interpreted as a keyword. A column named "array" would break things here.

```suggestion
    dst_cols = ', '.join(f'"{dst}"' for _, dst in columns)
```